fix unknown theme error

BilalG1 opened the pull request:
<!--

Make sure you've read the CONTRIBUTING.md guidelines: https://github.com/stack-auth/stack-auth/blob/dev/CONTRIBUTING.md

-->

<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Fixes unknown theme error by adding fallback to default theme and updates theme display names.
> 
>   - **Behavior**:
>     - In `route.tsx`, added fallback to `DEFAULT_EMAIL_THEME_ID` if current theme is unknown.
>     - Updates active theme in `tenancy.completeConfig.emails` if unknown.
>   - **Display Name Updates**:
>     - Changed `displayName` from `default-light` to `Default Light` and `default-dark` to `Default Dark` in `emails.ts` and `schema-fields.ts`.
>   - **Tests**:
>     - Updated test cases in `email-themes.test.ts` to reflect new display names.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=stack-auth%2Fstack-auth&utm_source=github&utm_medium=referral)<sup> for 8d2b9f75e2a1a23186cd3d10899092ff10c2cdb5. You can [customize](https://app.ellipsis.dev/stack-auth/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

N2D4 reviewed (changes requested):
Can we also get rid of the string `default-light` everywhere, rename it to `Default Light`, and not use display names as identifiers anywhere?

BilalG1 commented:
changed default-light and default-dark to Default Light and Default Dark. removed using display name identifiers here: https://github.com/stack-auth/stack-auth/pull/785/files
